### PR TITLE
[wip] logging change to try and figure out lb issue

### PR DESCRIPTION
**Pull request by squeed**

### CODE REVIEW

**Comment by squeed** [comment]
Checked with the CI cluster; this one came up fine.

**Comment by squeed** [comment]
ci issue
/test e2e-gcp

**Comment by squeed** [comment]
Once again, the e2e-gcp run does not suffer this issue (though crio was deadlocked for unrelated reasons, fun).

**Comment by squeed** [comment]
Green
/test e2e-gcp

File: vendor/k8s.io/client-go/tools/cache/controller.go [hunk 1]
```diff
@@ -357,7 +357,7 @@ func newInformer(
 			// from oldest to newest
 			for _, d := range obj.(Deltas) {
 				switch d.Type {
-				case Sync, Added, Updated:
+				case Sync, Added, Updated, Replaced:
```
Comment (sttts) [inline vendor/k8s.io/client-go/tools/cache/controller.go]: nit: Sync, Replace, Added, Updated

File: vendor/k8s.io/client-go/tools/cache/delta_fifo.go [hunk 2]
```diff
@@ -584,10 +584,10 @@ const (
 	Added   DeltaType = "Added"
 	Updated DeltaType = "Updated"
 	Deleted DeltaType = "Deleted"
-	// The other types are obvious. You'll get Sync deltas when:
-	//  * A watch expires/errors out and a new list/watch cycle is started.
-	//  * You've turned on periodic syncs.
-	// (Anything that trigger's DeltaFIFO's Replace() method.)
+	// Replaced is emitted when we encountered watch errors and had to do a
+	// relist. We don't know if the newly relisted object has changed.
```
Comment (sttts) [inline vendor/k8s.io/client-go/tools/cache/delta_fifo.go]: ... shared informers will compare against their store to find out.

File: vendor/k8s.io/client-go/tools/cache/shared_informer.go [hunk 3]
```diff
@@ -452,10 +453,14 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	// from oldest to newest
 	for _, d := range obj.(Deltas) {
 		switch d.Type {
-		case Sync, Added, Updated:
+		case Sync, Added, Updated, Replaced:
 			isSync := d.Type == Sync
 			s.cacheMutationDetector.AddObject(d.Object)
 			if old, exists, err := s.indexer.Get(d.Object); err == nil && exists {
+				// If the delta is from a relist, and the RV hasn't changed, elide it.
+				if d.Type == Replaced && isObjectSame(old, d.Object) {
```
Comment (sttts) [inline vendor/k8s.io/client-go/tools/cache/shared_informer.go]: this is an optimization. Better move it into its own commits, up for discussion.
